backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendDiskSnapshotsResource.java

```diff
@@ -40,14 +40,24 @@ protected DiskSnapshots mapCollection(List<org.ovirt.engine.core.common.business
             diskSnapshot.setDisk(new Disk());
             diskSnapshot.getDisk().setId(this.diskId.toString());
             collection.getDiskSnapshots().add(addLinks(populate(diskSnapshot, disk), Disk.class));
-            diskSnapshot.setHref(buildHref(diskId.toString(), diskSnapshot.getId().toString()));
+            diskSnapshot.setHref(buildHref(diskId.toString(), diskSnapshot.getId()));
             if (diskSnapshot.getParent() != null) {
-                diskSnapshot.getParent().setHref(buildParentHref(diskId.toString(), false));
+                org.ovirt.engine.core.common.businessentities.storage.Disk
+                        parentDisk = getDiskIdFromDiskSnapshotId(entities, diskSnapshot);
+                diskSnapshot.getParent().setHref(buildHref(parentDisk.getId().toString(),
+                        diskSnapshot.getParent().getId()));
             }
         }
         return collection;
     }
 
+    private org.ovirt.engine.core.common.businessentities.storage.Disk getDiskIdFromDiskSnapshotId
```

Comment:
This method is duplicated in the two classes. It does not access any fields, so I would make it static and extract it to some utils class (e.g DiskResourceUtils)